launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchUtils.java

```diff
@@ -82,4 +92,52 @@ public static final String toDockerVolume(Map<String, String> pMap, IPath path)
 		return rv;
 	}
 
+	/**
+	 * Pull an Docker-Image if is not locally available.
+	 *
+	 * The interface is rather bad, as it returns null on success to keep things simple at the call site.
```

Comment:
Why not have it return an IStatus?  For an error, you can grab the status message, otherwise it returns Ok status.

Reply:
This was basically: Mark code -> Extract to method (or whatever the command is) to make it testable. :) Thus I tried to no add additional logic.

Reply:
Done